crates/subspace-networking/src/bin/subspace-bootstrap-node/main.rs

```diff
@@ -220,3 +229,55 @@ async fn main() -> Result<(), Box<dyn Error>> {
 fn peer_id_from_keypair(keypair: Keypair) -> PeerId {
     peer_id(&libp2p::identity::Keypair::from(keypair))
 }
+
+/// Networking manager implementation for DSN bootstrap node.
+#[derive(Clone, Default)]
+pub(crate) struct NetworkingParametersManager {
+    address_removed: Handler<PeerAddressRemovedEvent>,
+}
+
+impl NetworkingParametersManager {
+    pub fn new() -> Self {
+        Self {
+            address_removed: Default::default(),
+        }
+    }
+    /// Returns an instance of `NetworkingParametersManager` as the `Box` reference.
+    pub fn boxed(self) -> Box<dyn NetworkingParametersRegistry> {
+        Box::new(self)
+    }
+}
+
+#[async_trait]
+impl NetworkingParametersRegistry for NetworkingParametersManager {
+    async fn add_known_peer(&mut self, _: PeerId, _: Vec<Multiaddr>) {}
+
+    async fn remove_known_peer_addresses(&mut self, peer_id: PeerId, addresses: Vec<Multiaddr>) {
+        for addr in addresses {
+            self.address_removed.call_simple(&PeerAddressRemovedEvent {
+                peer_id,
+                address: addr.clone(),
+            });
+        }
+    }
```

Comment:
Why is it removing address immediately comparing to what the other implementation that tracks duration since the peer was last time online?

In fact the name of this method is probably misleading and should be something like `notify_unreachable_address` or something.